kubectl-fdb/cmd/buggify_crash_loop_test.go

```diff
@@ -0,0 +1,215 @@
+					Expect(len(tc.ExpectedInstancesInCrashLoop)).To(BeNumerically("==", len(resCluster.Spec.Buggify.CrashLoop)))
+				},
+				Entry("Adding single instance.",
+					testCase{
+						Instances:                    []string{"instance-1"},
+						ExpectedInstancesInCrashLoop: []string{"instance-1"},
+					}),
+				Entry("Adding multiple instances.",
+					testCase{
+						Instances:                    []string{"instance-1", "instance-2"},
+						ExpectedInstancesInCrashLoop: []string{"instance-1", "instance-2"},
+					}),
+			)
+
+			When("a process group was already in crash-loop", func() {
+				var kubeClient client.Client
+
+				BeforeEach(func() {
+					cluster.Spec.Buggify.CrashLoop = []string{"instance-1"}
+					scheme := runtime.NewScheme()
+					_ = clientgoscheme.AddToScheme(scheme)
+					_ = fdbv1beta2.AddToScheme(scheme)
+					kubeClient = fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(cluster, &corev1.PodList{}).Build()
+				})
+
+				type testCase struct {
+					Instances                    []string
+					ExpectedInstancesInCrashLoop []string
+				}
+
+				DescribeTable("should add all targeted processes to crash-loop list",
+					func(tc testCase) {
+						err := updateCrashLoopList(kubeClient, clusterName, tc.Instances, namespace, false, false, false)
+						Expect(err).NotTo(HaveOccurred())
+
+						var resCluster fdbv1beta2.FoundationDBCluster
+						err = kubeClient.Get(ctx.Background(), client.ObjectKey{
+							Namespace: namespace,
+							Name:      clusterName,
+						}, &resCluster)
+						Expect(err).NotTo(HaveOccurred())
+						Expect(tc.ExpectedInstancesInCrashLoop).To(ContainElements(resCluster.Spec.Buggify.CrashLoop))
+						Expect(len(tc.ExpectedInstancesInCrashLoop)).To(BeNumerically("==", len(resCluster.Spec.Buggify.CrashLoop)))
+					},
+					Entry("Adding the same instance.",
+						testCase{
+							Instances:                    []string{"instance-1"},
+							ExpectedInstancesInCrashLoop: []string{"instance-1"},
+						}),
+					Entry("Adding different instance.",
+						testCase{
+							Instances:                    []string{"instance-2"},
+							ExpectedInstancesInCrashLoop: []string{"instance-1", "instance-2"},
+						}),
+					Entry("Adding multiple instances.",
+						testCase{
+							Instances:                    []string{"instance-2", "instance-3"},
+							ExpectedInstancesInCrashLoop: []string{"instance-1", "instance-2", "instance-3"},
+						}),
+				)
+			})
+		})
+
+		When("removing instances from crash-loop list from a cluster", func() {
+			var kubeClient client.Client
+
+			BeforeEach(func() {
+				cluster.Spec.Buggify.CrashLoop = []string{"instance-1", "instance-2", "instance-3"}
+				scheme := runtime.NewScheme()
+				_ = clientgoscheme.AddToScheme(scheme)
+				_ = fdbv1beta2.AddToScheme(scheme)
+				kubeClient = fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(cluster, &corev1.PodList{}).Build()
```

Comment:
Why are we adding here an empty Pod list if the `updateCrashLoopList` method doesn't query any Pods?